src/bigquery.ts

```diff
@@ -251,9 +251,14 @@ export class BigQuery extends common.Service {
     options = Object.assign({}, options, {
       apiEndpoint: options.apiEndpoint || 'bigquery.googleapis.com',
     });
+
+    const baseUrl =
+      process.env.BIGQUERY_EMULATOR_HOST ||
+      `https://${options.apiEndpoint}/bigquery/v2`;
```

Comment:
I've never been fully clear on the right way to go about this, and maybe each library is different-- should the env var replace everything, including the "/bigquery/v2" path? Or just the value of "options.apiEndpoint", etc.

Reply:
We want it to replace everything, which is also how it's done in [nodejs-storage](https://github.com/googleapis/nodejs-storage/blob/785b12f09919700c30106920ba3d4e578959591f/src/storage.ts#L403)

Reply:
We do a bit more there, too: https://github.com/googleapis/nodejs-storage/blob/d6dfdd244511d8b3ffaab2f561f272e8d81a2e8a/src/storage.ts#L431

- inserts a protocol if one is missing
- removes trailing slashes

It also treats the emulator env var as the `apiEndpoint` that gets passed to `super(config, ...)`, where this PR only uses it for `baseUrl`. Overall, it might be good to sync these two up, having this implementation match Storage.

Reply:
The reasons why they are divergent is that BigQuery expects that the provided `apiEndpoint` will not have a protocol and will not be manipulated before being set in the Service config. If changing the expected behavior for `apiEndpoint` is okay, I can go ahead and make it totally follow Storage.

Reply:
@stephenplusplus I went ahead and matched Storage's implementation. The major difference is that BigQuery assumed that apiEndpoint would be set on the client without a protocol. I'm thinking that changing this might be breaking to some if they are using this property for anything.

Reply:
Thanks. It looks like the new code will handle when a user provides an `apiEndpoint` without a protocol when it passes through the `sanitizeEndpoint` function. So, I don't think there's a breaking path there, but let me know if I missed it.

Reply:
@stephenplusplus What I mean is that if a user is used to initializing a client with `const client = new BigQuery({apiEndpoint: 'test.com'})` they would be used to `client.apiEndpoint` returning `'test.com'`, whereas now it will return `'https://test.com'`.

Reply:
Does that affect anything, in terms of the address that's built to make API requests with? Users shouldn't be reading the `apiEndpoint` property, that's an internal/only-we-care thing.

Reply:
No, it doesn't affect anything. I was just concerned that I was missing a reason why reading this property might matter to users, but sounds like it's totally not a concern and we're all good.

Reply:
It's great that you thought of that, I wouldn't have even considered that (...as was clear by my confusion). Sorry about that!